File: tests/dictionaries/string-dict.swift
public func newEmpty() -> [String: String] {
    return [:]
}

public func newSingle(key: String, value: String) -> [String: String] {
    return [key: value]
}

public func dictGet(_ dict: [String: String], key: String) -> String? {
    return dict[key]
}

public func dictSet(_ dict: inout [String: String], key: String, value: String) -> () {
    dict[key] = value
}

public func count(_ dict: [String: String]) -> Int {
    return dict.count
}

public func allKeys(_ dict: [String: String]) -> Dictionary<String, String>.Keys {
    return dict.keys
}

public func firstKey(_ dict: [String: String]) -> String? {
    return dict.keys.first
}

File: tests/dictionaries/string-dict.js
export function newEmpty() {
  return {};
}
export function newSingle$key$value$(key, value) {
  return {
    [key]: value
  };
}
export function dictGet$key$(dict, key) {
  return Object.hasOwnProperty.call(dict, key) ? dict[key] : null;
}
export function dictSet$key$value$(dict, key, value) {
  if (value !== null) {
    dict[key] = value;
  } else {
    delete dict[key];
  }
}
export function count(dict) {
  return Object.keys(dict).length;
}
export function allKeys(dict) {
  return Object.keys(dict);
}
export function firstKey(dict) {
  const keys = Object.keys(dict);
  return keys.length ? keys[0] : null;
}